chainercv/utils/iterator/apply_to_iterator.py

```diff
@@ -0,0 +1,168 @@
+from chainercv.utils.iterator.unzip import unzip
+
+
+def apply_to_iterator(func, iterator, n_input=1, hook=None):
+    """Apply a function/method to batches from an iterator.
+
+    This function applies a function/method to an iterator of batches.
+
+    It assumes that the iterator iterates over a collection of tuples
+    that contain inputs to :func:`func`.
+    Additionally, the tuples may contain values
+    that are not used by :func:`func`.
+    For convenience, we allow the iterator to iterate over a collection of
+    inputs that are not tuple.
+    Here is an illustration of the expected behavior of the iterator.
```

Comment:
I think this comment is helpful for Chainer users.

---

Here is an ....
In short, we require `iterator` to satisfy the interface of `chainer.Iterator`.

Reply:
`chainer.Iterator` requires many properties, such as `batch_size`, `epoch_detail`, etc..
`apply_to_iterator` does not require these properties.
https://docs.chainer.org/en/stable/reference/generated/chainer.iterators.SerialIterator.html

Reply:
How about `chainer.Iteratror satisfies this requirement` or  `chainer.Iteratror satisfies this behaviour`? These are correct.

Reply:
`This behaviour is same as chainer.Iterator` is also OK.

Reply:
This behaviour is the same as chainer.Iterator 

This one looks good! (“the” was missing)